packages/front-end/components/AutoGenerateFactTablesModal.tsx

```diff
@@ -0,0 +1,553 @@
+          setAutoFactTableError(res.message);
+          return;
+        }
+
+        if (res.autoFactTablesToCreate.length) {
+          form.setValue(
+            "factTablesToCreate",
+            res.autoFactTablesToCreate.map((table) => ({
+              ...table,
+              showSqlPreview: false,
+            }))
+          );
+        }
+      } catch (e) {
+        track("Generate Auto Fact Tables Error", {
+          error: e.message,
+          source,
+          type: datasourceObj.type,
+          dataSourceId: datasourceObj.id,
+          schema: datasourceObj.settings.schemaFormat,
+        });
+        setAutoFactTableError(e.message);
+      }
+    },
+    [apiCall, form, selectedSchema, source]
+  );
+
+  useEffect(() => {
+    if (!selectedDatasource) return;
+
+    if (!selectedSchema && availableSchemas.length === 1) {
+      form.setValue("schema", availableSchemas[0]);
+    }
+
+    if (!selectedSchema) return;
+
+    getTrackedEvents(selectedDatasource);
+  }, [
+    availableSchemas,
+    form,
+    getTrackedEvents,
+    selectedDatasource,
+    selectedSchema,
+  ]);
+
+  useEffect(() => {
+    if (selectedDatasourceData?.informationSchema) {
+      const schemas: string[] = [];
+      selectedDatasourceData.informationSchema.databases.forEach((database) => {
+        database.schemas.forEach((schema) => {
+          schemas.push(schema.schemaName);
+        });
+      });
+      setAvailableSchemas(schemas);
+    }
+  }, [selectedDatasourceData?.informationSchema]);
+
+  useEffect(() => {
+    if (refreshingSchema) {
+      if (
+        retryCount > 1 &&
+        retryCount < 8 &&
+        selectedDatasourceData?.informationSchema?.status === "COMPLETE"
+      ) {
+        setRefreshingSchema(false);
+        setRetryCount(1);
+      } else if (retryCount > 8) {
+        setRefreshingSchema(false);
+        setRefreshingSchemaError(
+          "This query is taking quite a while. We're building this in the background. Feel free to leave this page and check back in a few minutes."
+        );
+        setRetryCount(1);
+      } else {
+        const timer = setTimeout(() => {
+          mutate();
+          setRetryCount(retryCount * 2);
+        }, retryCount * 1000);
+        return () => {
+          clearTimeout(timer);
+        };
+      }
+    }
+  }, [
+    refreshingSchema,
+    mutate,
+    retryCount,
+    selectedDatasourceData?.informationSchema?.status,
+  ]);
+
+  useEffect(() => {
+    async function getInformationSchema(dataSourceId: string) {
+      const { informationSchema } = await apiCall<{
+        status: number;
+        informationSchema?: InformationSchemaInterface;
+      }>(`/datasource/${dataSourceId}/schema`, {});
+      if (informationSchema?.error?.message) {
+        setAutoFactTableError(informationSchema.error.message);
+      }
+      setSelectedDatasourceData({ informationSchema });
+    }
+
+    if (selectedDatasource?.id) {
+      getInformationSchema(selectedDatasource.id);
```

Comment:
Need to add a `.catch` here, otherwise, this will be an uncaught error if the API call fails